### PR TITLE
pubmedqa-env added

### PR DESCRIPTION
**Pull request by rscgh**
Here is my implementation of the pubmedqa benchmark (Issue #5). 
* it uses the test set as defined in the original [pubmedqa github repo](https://github.com/pubmedqa/pubmedqa/blob/master/data/test_ground_truth.json)
* as the bigbio repo uses the old-style dataset format (non-parquet but zip based), I opted for using the dataset from the following repo: https://huggingface.co/datasets/qiaojin/PubMedQA (but can change to bigbio if desired)
* I tried to keep the logics close to that of InspectEvals (for comparability), but they might have an [issue with prompt formatting](https://github.com/UKGovernmentBEIS/inspect_evals/pull/479#issuecomment-3298790361).
* The current implementation defines no system prompt, and keeps close to Figure 1 of the [original pubmedqa paper](https://arxiv.org/pdf/1909.06146) (with the main difference is that in the current implementation, the question comes after the context/paper abstract, see the readme for an example).
* After implementation, I also discovered a separate [pubmedqa implementation on primeintellect](https://app.primeintellect.ai/dashboard/environments/maziyar/openmed_pubmedqa/files/mp7jwptm9omenhdw7pneb1wn/OpenMed_PubMedQA.py), but this seems to use only a subset of 200 questions (instead of the full 500) and prompts for thinking explicitly.

### CODE REVIEW

**Comment by tmabraham** [comment]
@rscgh this is awesome! just one minor thing...

**Comment by rscgh** [comment]
The hf dataset I was using only had the test split. I reuploaded the entire dataset with both splits and updated the code. 
Also it seems like the InspectEval contributer has reached out to the PubMedQA authors to inquire about the correct prompt format, so let's see if they'll get a reply.

**Comment by tmabraham** [comment]
@rscgh I'm a bit confused, how is this split different from what you already had? You just set the training set to `pqaa` but the original dataset already had `pqa_artificial`

**Comment by warner-benjamin** [comment]
Looks like the eval harness isn't the only standard. We can follow the [helm version](https://github.com/huggingface/lighteval/blob/ecef2c662b9418866b6447d33b5e7d5dedd74af8/src/lighteval/tasks/default_prompts.py#L2087) which uses "Answer A for yes, B for no or C for maybe" which is probably better than the [multiple-choice test close](https://github.com/huggingface/lighteval/blob/ecef2c662b9418866b6447d33b5e7d5dedd74af8/tests/tasks/templates/test_multichoice.py#L64-L65) expecting the model to output only yes, no, or maybe.
